### PR TITLE
Anna Melnyk ІМ-32 Winter-2024-Exam

### PR DESCRIPTION
**Pull request by ani163**
Done: 
intersection.js
invert.js
password.js
random.js
reverse.js
sample.js
split.js
types.js
words.js
zip.js

👾

### CODE REVIEW

File: Tasks/intersection.js [hunk 1]
```diff
@@ -1,25 +1,19 @@
...
-      }
-    }
-    {
+const findIntersection = (object1, object2) => {
+  const keys1 = Object.keys(object1);
+  const intersection = object1;
+
+  for (const key of keys1) {
+    if (object1[key] !== object2[key]) {
+      delete intersection[key];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:12]: Try to implement without delete

File: Tasks/password.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+const generatePassword = (alphabet, length) => {
+  let password = '';
 
-let GeneratePassword = (alphabet, length) => {
-  const MAX = alphabet.length;
-  key = '';
   for (let i = 0; i < length; i++) {
-    Index = Math.floor(Math.random() * MAX);
-    key = key + alphabet[Index];
+    const randomIndex = Math.floor(Math.random() * alphabet.length);
```
Comment (tshemsedinov) [inline Tasks/password.js:10]: Try to decompose into two functions

File: Tasks/random.js [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
-Random = (min, max) => {
-  if (max === undefined) {
-    max = min;
-    return Math.floor(Math.random() * (max + 1));
-  } else {
-    return min + Math.floor(Math.random() * (max - min + 1));
-  }
+const getRandomInt = (min, max = min) => {
+  const range = max - min + 1;
+  return min + Math.floor(Math.random() * range);
```
Comment (tshemsedinov) [inline Tasks/random.js:8]: Also try to decompose

File: Tasks/sample.js [hunk 4]
```diff
@@ -1,8 +1,11 @@
+'use strict';
+
 // Get one random element from an array
+// Fixed
 
-const SAMPLE = (arr) => {
-  arr = arr[Math.floor(Math.random() * arr.length)];
-  return arr;
+const getRandomElement = (arr) => {
+  const randomIndex = Math.floor(Math.random() * arr.length);
```
Comment (tshemsedinov) [inline Tasks/sample.js:7]: Ditto

File: Tasks/types.js [hunk 5]
```diff
@@ -1,18 +1,21 @@
...
+// Fixed
 
-types_ = function (s) {
-  types_ = {
+const countTypes = (arr) => {
+  const types = {
     number: 0,
     string: 0,
     boolean: 0,
   };
```
Comment (tshemsedinov) [inline Tasks/types.js:11]: We may generate keys for types dynamicaly

File: Tasks/words.js [hunk 6]
```diff
@@ -1,41 +1,19 @@
...
-      } else {
-        if (flag === true) {
-          flag = true;
-        } else {
-          flag = true;
-        }
-      }
+  for (const word of wordsArray) {
+    if (word !== '') {
+      numberOfWords++;
```
Comment (tshemsedinov) [inline Tasks/words.js:12]: Single line will be ok here

File: Tasks/zip.js [hunk 7]
```diff
@@ -1,18 +1,21 @@
...
-  j = 0;
-  for (x of b) {
-    CELL = [a[i++], x];
-    if (i < j) {
-      delete a[i++];
-    } else {
-      (() => (b[j++] = CELL))();
+
+  for (const elem of arr2) {
+    const cell = [arr1[i++], elem];
```
Comment (tshemsedinov) [inline Tasks/zip.js:11]: Try more consistent way of iteration

File: Tasks/intersection.js (hunk 1 above)
Comment (ani163) [reply Tasks/intersection.js]: I tried something without delete, but I'm not sure that it's what you asked about

File: Tasks/password.js (hunk 2 above)
Comment (ani163) [reply Tasks/password.js]: Done!

File: Tasks/random.js (hunk 3 above)
Comment (ani163) [reply Tasks/random.js:12]: Also done

File: Tasks/sample.js (hunk 4 above)
Comment (ani163) [reply Tasks/sample.js]: 🫡

File: Tasks/types.js (hunk 5 above)
Comment (ani163) [reply Tasks/types.js:11]: Now keys for types may be generated dynamicaly

File: Tasks/words.js (hunk 6 above)
Comment (ani163) [reply Tasks/words.js]: Changed it

File: Tasks/zip.js (hunk 7 above)
Comment (ani163) [reply Tasks/zip.js]: Done! Hope new version is better

**Pull request closed without merging**